airflow/bin/cli.py

```diff
@@ -1116,6 +1129,34 @@ def kerberos(args):  # noqa
         airflow.security.kerberos.run()
 
 
+def create_admin(args):
```

Comment:
Does this work if something if FAB is configured to use other auth backends than AUTH_DB?

Reply:
@ashb The create_admin is for auth_db only. The OAuth flow (which I tested) will also create an entry in the ab_user table, with a username assembled by FAB (provider name plus an id it parsed from the auth response). So if you know the username and enter with provide prefix, you could create an account via the command, but it's not very obvious.

I spent a bit of time on sorting out the OAuth flow today. The registration is best done via the UI and is a bit more involved. It requires the following config change
1. Modify webserver_config.py
```
AUTH_TYPE = AUTH_OAUTH
OAUTH_PROVIDERS = [
{
    'name':'google', 
    'whitelist': ['@COMPANY.COM'],
    'token_key':'access_token',
    'icon':'fa-google',
        'remote_app': {
            'base_url':'https://www.googleapis.com/oauth2/v2/',
            'request_token_params':{
                'scope': 'email profile'
            },
            'access_token_url':'https://accounts.google.com/o/oauth2/token',
            'authorize_url':'https://accounts.google.com/o/oauth2/auth',
            'request_token_url': None,
            'consumer_key': CONSUMER_KEY,
            'consumer_secret': SECRET_KEY,
        }
    }
]
AUTH_USER_REGISTRATION = True
AUTH_USER_REGISTRATION_ROLE = "Admin"
```
This will allow the admin to self register an account with full access
2. Once the admin(s) is registered, modify webserver_config.py again:
```
AUTH_USER_REGISTRATION_ROLE = "Public"
```
This will allow all other users to freely register with the `Public` role via OAuth. 

Will add a note about this in the doc.

Reply:
Just found out that FAB provides a nice way to override/customize the oauth registration stored in the db, which will make this create_admin cli tool more usable (for example, username can be set to email address), http://flask-appbuilder.readthedocs.io/en/latest/security.html#authentication-oauth details and link to example

Reply:
It sounds like we can configure FAB logins so this works for all backends, but if not is it worth guarding this function so that it says "can't be used with this auth backend" or a similar error (rather than silently creating a user record that won't do anything)

It also might be worth generalizing this to a `create_user` command instead, and adding the role/group name(s) as a parameter -- this would let people create users via the CLI too which sounds like a plus for scripting.

Reply:
correct, we can configure it such that this works for all backends. And great idea, will do.

Reply:
Are we going to do this or not?

Reply:
@ashb  oops, yes! this slipped my mind hehe. working on this

Reply:
done.